src/MicrosoftAi/AbstractionMapper.cs

```diff
@@ -0,0 +1,321 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Extensions.AI;
+using OllamaSharp.Models.Chat;
+
+namespace OllamaSharp.MicrosoftAi;
+
+/// <summary>
+/// Provides mapping functionality between OllamaSharp and Microsoft.Extensions.AI models.
+/// </summary>
+public static class AbstractionMapper
```

Comment:
The mapping is not finished yet. If anyone knows the missing details how to implement my TODOs, I'd be happy to hear them.

Reply:
Can this type be `internal`? Not sure why app developers would need to call the mapping methods.

Reply:
Right now it's for testing purposes only. I placed it into another namespace at least. I tried `InternalVisibleTo` without creating an assembly info but it did not work. Might give it another try.